expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/extension/OperationToRequestExtension.kt

```diff
@@ -0,0 +1,151 @@
+package com.expediagroup.sdk.rest.extension
+
+import java.io.IOException
+import java.net.MalformedURLException
+import java.net.URL
+
+import com.expediagroup.sdk.core.http.MediaType
+import com.expediagroup.sdk.core.http.Method
+import com.expediagroup.sdk.core.http.Request
+import com.expediagroup.sdk.core.http.RequestBody
+
+import com.expediagroup.sdk.rest.trait.operation.ContentTypeTrait
+import com.expediagroup.sdk.rest.trait.operation.HeadersTrait
+import com.expediagroup.sdk.rest.trait.operation.HttpMethodTrait
+import com.expediagroup.sdk.rest.trait.operation.OperationRequestBodyTrait
+import com.expediagroup.sdk.rest.trait.operation.OperationRequestTrait
+import com.expediagroup.sdk.rest.trait.operation.UrlPathTrait
+import com.expediagroup.sdk.rest.trait.operation.UrlQueryParamsTrait
+import com.expediagroup.sdk.rest.trait.serialization.SerializeRequestBodyTrait
+
+/**
+ * Extension function to parse an operation request into an HTTP request.
+ *
+ * This function takes an operation request and converts it into an HTTP request
+ * by setting the HTTP method, headers, body, and URL based on the traits of the operation.
+ *
+ * @param serverUrl the base server URL
+ * @param serializer instance used to serialize the request body to an InputStream
+ * @return the constructed HTTP request
+ * @throws IllegalArgumentException if the request body is invalid
+ * @throws IllegalStateException if the HTTP method or URL is not set
+ * @throws IOException if an I/O error occurs
+ */
+@Throws(
+    IllegalArgumentException::class,
+    IllegalStateException::class,
+    IOException::class
+)
+internal fun OperationRequestTrait.parseRequest(
+    serverUrl: URL,
+    serializer: SerializeRequestBodyTrait
+): Request {
+    require(this is HttpMethodTrait && this.getHttpMethod().isNotBlank()) {
```

Comment:
So this should actually be an extension of an interface that implements OperationRequestTrait and HttpMethodTrait

Reply:
Please check [this](https://github.com/ExpediaGroup/lodging-connectivity-java-sdk/pull/166/files#r1938443782) comment.